File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnResponseError.java

package com.gracenote.gnsdk;

public class GnResponseError extends GnDataObject {
  private transient long swigCPtr;

  protected GnResponseError(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnResponseError_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnResponseError obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnResponseError(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnResponseError_gnType();
  }

  public static GnResponseError from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnResponseError(gnsdk_javaJNI.GnResponseError_from(GnDataObject.getCPtr(obj), obj), true);
  }

  public GnResponseError() {
    this(gnsdk_javaJNI.new_GnResponseError(), true);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnResponseError.cs

namespace GracenoteSDK {

public class GnResponseError : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnResponseError(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnResponseError_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnResponseError obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnResponseError() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnResponseError(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnResponseError_GnType();
    return ret;
  }

  public static GnResponseError From(GnDataObject obj) {
    GnResponseError ret = new GnResponseError(gnsdk_csharp_marshalPINVOKE.GnResponseError_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public GnResponseError() : this(gnsdk_csharp_marshalPINVOKE.new_GnResponseError(), true) {
  }

}

}
